src/main/java/org/squiddev/plethora/integration/hatchery/IntegrationHatchery.java

```diff
@@ -0,0 +1,287 @@
+			if (entity == null)
+				entity = EntityList.createEntityByIDFromName(new ResourceLocation(entityData.getString("id")), location.getWorld());
+			if (entity == null) return getBasicDetails(entityData);
+
+			Vec3d loc = location.getLoc();
+			entity.setPositionAndRotation(loc.x, loc.y, loc.z, 0, 0);
+			return Collections.singletonMap("spawnEntity", context.makePartialChild(entity).getMeta());
+		}
+
+		//Copied wholesale from IntegrationThermalExpansion.META_MORB
+		private Map<Object, Object> getBasicDetails(NBTTagCompound entityData) {
+			String translationKey = EntityList.getTranslationName(new ResourceLocation(entityData.getString("id")));
+			if (translationKey == null) return Collections.emptyMap();
+
+			String translated = Helpers.translateToLocal("entity." + translationKey + ".name");
+
+			Map<Object, Object> details = new HashMap<>(2);
+			details.put("name", translated);
+			details.put("displayName",
+				entityData.hasKey("CustomName", Constants.NBT.TAG_STRING)
+					? entityData.getString("CustomName")
+					: translated
+			);
+			return Collections.singletonMap("spawnEntity", details);
+		}
+
+		@Nullable
+		@Override
+		public ItemStack getExample() {
+			//REFINE What are the trade-offs between this technique and `new ResourceLocation("minecraft:chicken")`?
+			ResourceLocation chickenLocation = EntityList.getKey(EntityChicken.class);
+			if (chickenLocation == null) return null;
+
+			//REFINE Similarly, how does this compare with `new EntityChicken(WorldDummy.INSTANCE)`
+			Entity chicken = EntityList.createEntityByIDFromName(chickenLocation, WorldDummy.INSTANCE);
+			if (!(chicken instanceof EntityAgeable)) return null;
+
+			return HatcheryEgg.createEggFromEntity(WorldDummy.INSTANCE, (EntityAgeable) chicken);
+		}
+	};
+
+	public static final IMetaProvider<EntityRooster> META_ROOSTER = new BaseMetaProvider<EntityRooster>() {
+
+		@Nonnull
+		@Override
+		public Map<Object, Object> getMeta(@Nonnull IPartialContext<EntityRooster> context) {
+			Map<Object, Object> out = new HashMap<>();
+			EntityRooster rooster = context.getTarget();
+
+			//Somewhat of a misnomer; this is the 'energy' level, comparable to a furnace's remaining burn ticks
+			//The actual 'temptation items' (Hatchery's term, not mine) is exposed via the Item Handler Capability,
+			//TODO How can I expose the Rooster's inventory in a manner consistent with the rest of the code?
+			// I can't bind an Introspection Module to it...
+			out.put("seeds", rooster.getSeeds());
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public EntityRooster getExample() {
+			//REFINE See META_HATCHERY_EGG#getExample's task
+			ResourceLocation roosterLocation = EntityList.getKey(EntityRooster.class);
+			if (roosterLocation == null) return null;
+
+			Entity rooster = EntityList.createEntityByIDFromName(roosterLocation, WorldDummy.INSTANCE);
+			if (!(rooster instanceof EntityRooster)) return null;
+
+			return (EntityRooster) rooster;
```

Comment:
Can we just construct an `EntityRooster` directly, without going through `EntityList`?

Reply:
Done, pending commit